File: src/main/java/com/beat/global/common/dto/ErrorResponse.java
package com.beat.global.common.dto;

import com.beat.global.common.exception.base.BaseErrorCode;

public record  ErrorResponse(
        int status,
        String message
) {
    public static ErrorResponse of(final int status, final String message) {
        return new ErrorResponse(status, message);
    }

    public static ErrorResponse of(final BaseErrorCode baseErrorCode) {
        return new ErrorResponse(baseErrorCode.getStatus(), baseErrorCode.getMessage());
    }
}

File: global-support/src/main/kotlin/com/beat/global/support/response/ErrorResponse.kt
package com.beat.global.support.response

data class ErrorResponse(
    val status: Int,
    val message: String,
) {
    companion object {
        @JvmStatic
        fun of(
            status: Int,
            message: String,
        ): ErrorResponse = ErrorResponse(status, message)
    }
}
